src/Network/Network/NatGateway/GetAzureRMNatGatewayCommand.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        private const string ListParameterSet = "ListParameterSet";
+        private const string GetByNameParameterSet = "GetByNameParameterSet";
+        private const string GetByResourceIdParameterSet = "GetByResourceIdParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource group name of the nat gateway.",
+            ParameterSetName = GetByNameParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The resource group name of the nat gateway.",
+            ParameterSetName = ListParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ResourceName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the nat gateway.",
+            ParameterSetName = GetByNameParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Network/natGateways", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Nat Gateway Id",
+            ParameterSetName = GetByResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string NatGatewayId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource reference to be expanded.",
+            ParameterSetName = "Expand",
+            ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ExpandResource { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.NatGatewayId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.NatGatewayId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+            if (!string.IsNullOrEmpty(this.Name))
+            {
+                var vNatGateway = this.NetworkClient.NetworkManagementClient.NatGateways.Get(ResourceGroupName, Name, ExpandResource);
+                var vNatGatewayModel = NetworkResourceManagerProfile.Mapper.Map<CNM.PSNatGateway>(vNatGateway);
+                vNatGatewayModel.ResourceGroupName = this.ResourceGroupName;
+                vNatGatewayModel.Tag = TagsConversionHelper.CreateTagHashtable(vNatGateway.Tags);
+                WriteObject(vNatGatewayModel, true);
+            }
+            else
+            {
+                IPage<NatGateway> vNatGatewayPage;
+                if(!string.IsNullOrEmpty(this.ResourceGroupName))
```

Comment:
Please change to ` if(ShouldListByResourceGroup(this.ResourceGroupName, this.Name))` to support wildcards properly